Subject: [PATCH] Untrusted app will crash because wifiscanner is protected by selinux.

Test: manually installs gms or third party to get
WIFI_SCANNING_SERVICE:
  context.getSystemService(Context.WIFI_SCANNING_SERVICE);

diff --git a/core/java/android/app/SystemServiceRegistry.java b/core/java/android/app/SystemServiceRegistry.java
index 4c5fb13..18bcad7 100644
--- a/core/java/android/app/SystemServiceRegistry.java
+++ b/core/java/android/app/SystemServiceRegistry.java
@@ -532,6 +532,9 @@
             public WifiScanner createService(ContextImpl ctx) {
                 IBinder b = ServiceManager.getService(Context.WIFI_SCANNING_SERVICE);
                 IWifiScanner service = IWifiScanner.Stub.asInterface(b);
+                if (service == null) {
+                    return null;
+                }
                 return new WifiScanner(ctx.getOuterContext(), service,
                         ConnectivityThread.getInstanceLooper());
             }});
